### ISSUE TITLE
`pct new --info <template>` should work with namespaced directory structure

### ISSUE THREAD

**Issue by da-ar** [puppetlabs/pdkgo#200]
With the updates to a namespaced directory structure `pct new --info <template>` no longer works.

- [x] It should be able to return template info for `author/id`
- [x] It should autocomplete `author/id` -e.g
```
pct new --list pu <TAB>
> puppetlabs/
pct new --list puppetlabs/puppet-m <TAB>
> puppetlabs/puppet-module-base
```

**Assumptions:**

- The latest template version info will be returned

**Comment by michaeltlombardi** [comment]
Doing some preliminary work on this and discovered a few things:

1. We likely want to cache the full list of templates right away so we can slice them whether we're acting on one or all - especially since we sometimes slice and sometimes read by file path separately; I'm not sure the performance hit justifies the complexity of implementation vs just returning the list of templates and using that
2. The `pct new` command does not currently have a way to distinguish between templates with the same name by different authors, nevermind versions - I think we _can_ punt the versions for now and always use latest, but we need to resolve the author-level namespacing
3. I'm peering around to get a better understanding of when we call `Get` vs `GetInfo` since `Get` just seems to be a wrapper around `GetInfo`

I think we could keep the method call the same (ie only specify template cache path and id) but we would then _have_ to either select the first one found (alphabetically by author, so `bar/baz` instead of `foo/baz`) or error/whatever. Similarly, if we're not going to specify version, we will need to infer that too.

I am circling around the idea of proposing:

1. That we cache the template listing at the beginning of a `new` call regardless of if you're looking for one template or all templates; in fact, `List` already supports specifying a single template
2. That we add the `--TemplateAuthor` and `--TemplateVersion` flags which allow override of the default (first alphabetical, highest version) as noted above
3. That the filtering for latest version only be placed behind a flag on `List` like `LatestOnly` which is defaulted to true when specifying the `--list` flag but not others - this will allow us to have the full cached list of templates at the start and winnow them down as needed

Basically, I think we're bumping right up against some structural problems with our design which were only created (or at least made apparent) when we pivoted to namespaced templates. Now that you can have a template id with both multiple authors and versions, we need to handle that a little more robustly.

**Comment by sanfrancrisko** [comment]
I'd agree with your proposal. One thing I would perhaps differ on slightly is the `LatestOnly` flag. Although I agree with the default behaviour listing the latest version only, I think that perhaps the flag should be `AllVersions` (_or something along those lines_) if we want to list all versions available, rather than `LatestOnly=false`.

**Comment by michaeltlombardi** [comment]
Incidentally resolved by #202
